interfaces/prometheus_scrape/v0/schema.py

```diff
@@ -0,0 +1,140 @@
+"""
+This file defines the schemas for the provider and requirer sides of the `prometheus_scrape` interface.
+
+It exposes two interfaces.schema_base.DataBagSchema subclasses called:
+- ProviderSchema
+- RequirerSchema
+
+Examples:
+    ProviderSchema:
+        application_data: {
+            "alert_rules": {
+                "groups": [
+                    {
+                        "name": "an_alert_rule_group",
+                        "rules": [
+                            {
+                                "alert": "SomethingIsUp",
+                                "expr": "something_bad == 1",
+                                "for_": "0m",
+                                "labels": {
+                                    "some-label": "some-value"
+                                },
+                                "annotations": {
+                                    "some-annotation": "some-other-value"
+                                }
+                            }
+                        ]
+                    }
+                ]
+            },
+            "scrape_jobs": [
+                {
+                    "metrics_path": "/metrics",
+                    "static_configs": [
+                        {
+                            "targets": [
+                                "*:4080"
+                            ]
+                        }
+                    ]
+                }
+            ],
+            "scrape_metadata": {
+                "model": "cos",
+                "model_uuid": "c2e9f4d5-dcb3-4870-8509-330eb9745ee8",
+                "application": "zinc-k8s",
+                "unit": "zinc-k8s/0",
+                "charm_name": "zinc-k8s"
+            }
+        },
+        related_units: {
```

Comment:
this is a show-unit key, not in providerschema

perhaps we should add a separate example for the RequirerSchema containing what's here?

Reply:
ok so what confused me here is that in juju-status, the `related_units` key contains the 'other side' of the databag.

So I think it's fine if we replace this with:

```
unit:  {
                    "prometheus_scrape_unit_address": "zinc-k8s-0.zinc-k8s-endpoints.cos.svc.cluster.local",
                    "prometheus_scrape_unit_name": "zinc-k8s/0"
                }
    
```